File: src/main/java/com/trading/kalyani/KTManager/entity/DailyJobPlanner.java
package com.trading.kalyani.KTManager.entity;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "DailyJobPlanner")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class DailyJobPlanner {

    @EmbeddedId
    private DailyJobPlannerId id;

    private LocalDate jobForExpiryDate;

    private boolean isJobRequired;

    private LocalDateTime lastModified;

}

File: kpn-ui/src/app/models/dailyJobPlanner.ts
export interface DailyJobPlannerId {
    jobDate: string; // ISO date string
    appJobConfigNum: number;
  }
  
  export interface DailyJobPlanner {
    id: DailyJobPlannerId;
    jobForExpiryDate: string; // ISO date string
    lastModified: string; // ISO date string
    jobRequired: boolean;
  }